code/modules/plumbing/_the_tank.dm

```diff
@@ -0,0 +1,132 @@
+var/obj/machinery/plumbing_tank/master_plumber //Cool name, right?
+
+/proc/plumbing_has_reagents(volume)
+	return master_plumber && master_plumber.reagents.total_volume >= volume
+
+//The master reservoir connected to all plumbing on the station. Synthesizes power into reagents.
+/obj/machinery/plumbing_tank
+	name = "master plumbing tank"
+	desc = "A massive piece of industrial machinery covered in dials and gauges, used to control plumbing across the station."
+	icon = 'icons/obj/machines/mining_machines.dmi' //Temporary sprite
+	icon_state = "stacker"
+	obj_integrity = 500
+	max_integrity = 500
+	armor = list(melee = 60, bullet = -25, laser = 100, energy = 100, bomb = 80, bio = 100, rad = 100, fire = 100, acid = 100) //This thing is heavy-duty!
+	density = TRUE
+	anchored = TRUE
+	opacity = TRUE
+	idle_power_usage = 10
+	active_power_usage = 100
+	var/list/starting_reagents = list("water" = 1000)
+	var/operating_mode = PLUMBING_TANK_IDLE
+	var/obj/item/weapon/reagent_containers/buffer //Used as a guideline. The master tank will try to match the proportion of reagents in this.
+
+/obj/machinery/plumbing_tank/Initialize()
+	buffer = new/obj/item/weapon/reagent_containers/glass/bottle(src)
+	create_reagents(10000)
+	reagents.add_reagent_list(starting_reagents)
+	buffer.reagents.add_reagent_list(reagent_proportions_to_volume(reagents.get_reagent_proportions(), buffer.volume))
+	//Breaking down this line step-by-step!
+	//1. First we get the reagent proportions of the master tank. If we have 1234 water and 400 hydrogen and account for the empty space, this will be list("water" = 0.1234, "hydrogen" = 0.04).
+	//2. Next, we make a list (check holder.dm for both of these procs, by the way) of these proportions adjusted to the volume of our buffer.
+	//  If our bottle holds 30 units, this will be list("water" = 3.702, "hydrogen" = 1.2).
+	//3. Finally, add all those reagents to our buffer. Done!
+	//Keep in mind that we're accounting for empty space here. If there's 1 unit of water in a 1,000 unit tank, and it doesn't account for space, the water's proportion is 1, not 0.001!
+	if(!master_plumber)
+		master_plumber = src
+	..()
+
+/obj/machinery/plumbing_tank/Destroy()
+	if(master_plumber == src)
+		master_plumber = null
+	return ..()
+
+/obj/machinery/plumbing_tank/process()
+	switch(operating_mode)
+		if(PLUMBING_TANK_IDLE)
+			use_power = 1
+		if(PLUMBING_TANK_SYNTHESIZE)
+			use_power = 2
+			reagents.add_reagent_list(buffer.reagents.get_reagent_proportions())
+		if(PLUMBING_TANK_DRAIN)
+			use_power = 2
+			for(var/reagent in reagents.reagent_list)
+				var/datum/reagent/R = reagent
+				reagents.remove_reagent(R.id, 1)
+
+/obj/machinery/plumbing_tank/interact(mob/user)
+	var/dat
```

Comment:
Maybe I was not in the chat at the time this standard was made, same with Xhuis! Most of us which missed that discussion will only find out when RR comes to correct us!